pkgs/by-name/qq/qqmusic/package.nix

```diff
@@ -29,113 +34,71 @@
-# Mostly based on qqmusic-bin package from AUR:
-# https://aur.archlinux.org/packages/qqmusic-bin
-################################################################################
-stdenv.mkDerivation (finalAttrs: {
+let
   pname = "qqmusic";
-  version = "1.1.8";
-  src = fetchurl {
-    url = "https://c.y.qq.com/cgi-bin/file_redirect.fcg?bid=dldir&file=ecosfile_plink%2Fmusic_clntupate%2Flinux%2Fother%2Fqqmusic_${finalAttrs.version}_amd64.deb&sign=1-d1ca4d5c5a8369b26af88e881ba3ac544066a899dcaea29778b35c9f648e6fee-68cb7c1c";
-    name = "qqmusic.deb";
-    hash = "sha256-QtGNaow8F0FOW228DDrIk7slQMHFwJzpDSQYQ8xZN4g=";
-  };
-
-  nativeBuildInputs = [
-    autoPatchelfHook
-    makeWrapper
-    copyDesktopItems
-    dpkg
-  ];
-
-  buildInputs = [
-    alsa-lib
-    at-spi2-atk
-    at-spi2-core
-    cairo
-    cups
-    dbus
-    expat
-    gdk-pixbuf
-    glib
-    gtk3
-    libdbusmenu
-    libglvnd
-    libpulseaudio
-    mesa
-    nspr
-    nss
-    pango
-    pciutils
-    udev
-    xorg.libX11
-    xorg.libxcb
-    xorg.libXcomposite
-    xorg.libXcursor
-    xorg.libXdamage
-    xorg.libXext
-    xorg.libXfixes
-    xorg.libXi
-    xorg.libXrandr
-    xorg.libXrender
-    xorg.libXScrnSaver
-    xorg.libXtst
-  ];
-
-  unpackPhase = ''
-    runHook preUnpack
-
-    dpkg -x $src .
-
-    runHook postUnpack
-  '';
-
-  installPhase = ''
-    runHook preInstall
-
-    mkdir -p $out
-    cp -r opt/qqmusic $out/opt
-    cp -r usr/* $out/
-
-    rm -rf $out/opt/swiftshader
-    ln -sf ${libglvnd}/lib $out/opt/swiftshader
-
-    mkdir -p $out/bin
-    makeWrapper $out/opt/qqmusic $out/bin/qqmusic \
-      --argv0 "qqmusic" \
-      --add-flags "--no-sandbox" \
-      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath finalAttrs.buildInputs}"
-
-    runHook postInstall
-  '';
-
-  desktopItems = [
-    (makeDesktopItem {
-      name = "qqmusic";
-      desktopName = "QQMusic";
-      exec = "qqmusic %U";
-      terminal = false;
-      icon = "qqmusic";
-      startupWMClass = "qqmusic";
-      comment = "Tencent QQMusic";
-      categories = [ "AudioVideo" ];
-      extraConfig = {
-        "Name[zh_CN]" = "QQ音乐";
-        "Name[zh_TW]" = "QQ音樂";
-        "Comment[zh_CN]" = "腾讯QQ音乐";
-        "Comment[zh_TW]" = "騰訊QQ音樂";
-      };
-    })
-  ];
 
   meta = {
-    maintainers = with lib.maintainers; [ xddxdd ];
+    maintainers = with lib.maintainers; [
+      prince213
+      xddxdd
+    ];
     description = "Tencent QQ Music";
     homepage = "https://y.qq.com/";
-    platforms = [ "x86_64-linux" ];
+    platforms = [
+      "aarch64-darwin"
+      "x86_64-darwin"
+      "x86_64-linux"
+    ];
     license = lib.licenses.unfree;
     sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
     mainProgram = "qqmusic";
   };
-})
+in
+if stdenvNoCC.hostPlatform.isDarwin then
+  import ./darwin.nix {
+    inherit
+      meta
+      pname
+
+      fetchurl
+      stdenvNoCC
+      undmg
+      ;
+  }
+else
+  import ./linux.nix {
```

Comment:
I don't think that is cross-safe. We should use callPackage here, too and not inherit individual packages.